File: app/src/main/java/com/example/zdrowie/ui/kcal/KcalFragment.java
package com.example.zdrowie.ui.kcal;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.example.zdrowie.databinding.FragmentKcalBinding;

import java.text.DecimalFormat;

public class KcalFragment extends Fragment {

    private FragmentKcalBinding binding;

    private static final DecimalFormat decimalFormat = new DecimalFormat("0.0");

    private double height = 0.0;
    private double weight = 0.0;
    private int age = 0;
    private String sex = "";
    double kcal = 0.0;


    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        binding = FragmentKcalBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        EditText heightEditText =  (EditText) binding.heightEditNumber;
        heightEditText.addTextChangedListener(heightEditTextWatcher);

        EditText weightEditText =
                (EditText) binding.weightEditNumber;
        weightEditText.addTextChangedListener(weightEditTextWatcher);

        EditText ageEditText =
                (EditText) binding.ageEditNumber;
        ageEditText.addTextChangedListener(ageEditTextWatcher);

        RadioGroup sexRadioGroup = binding.sexRadioGroup;
        sexRadioGroup.setOnCheckedChangeListener((group, checkedId) -> {

            RadioButton checkedButton = (RadioButton) requireView().
                    findViewById(sexRadioGroup.getCheckedRadioButtonId());

            String checkedButtonText = checkedButton.getText().toString();

            switch (checkedButtonText) {
                case "Female":
                    sex = "female";
                    break;

                case "Male":
                    sex = "male";
                    break;
            }

            calculate();
        });

        return root;
    }

    private void calculate() {
        if ( weight > 0 && height > 0 && age > 0 && (sex.equals("female") || sex.equals("male"))) {
            if(sex.equals("male")) {
                kcal = 66.47 + (13.7 * weight) + (5.0 * height) - (6.76 * age);
            }
            else {
                kcal = 655.1 + (9.567 * weight) + (1.85 * height) - (4.68 * age);
            }
        }

        setText(decimalFormat.format(kcal));
    }

    public void setText(String text) {
        TextView view = (TextView) binding.bmrValueTextView;
        view.setText(text);
    }

    private final TextWatcher heightEditTextWatcher = new TextWatcher() {
        @Override
        public void onTextChanged(CharSequence s, int start,
                                  int before, int count) {
            try {
                height = Double.parseDouble(s.toString());
            } catch (NumberFormatException e) {
                height = 0.0;
            }

            calculate();
        }

        @Override
        public void afterTextChanged(Editable s) { }

        @Override
        public void beforeTextChanged(
                CharSequence s, int start, int count, int after) { }
    };

    private final TextWatcher weightEditTextWatcher = new TextWatcher() {
        @Override
        public void onTextChanged(CharSequence s, int start,
                                  int before, int count) {
            try {
                weight = Double.parseDouble(s.toString());
            }
            catch (NumberFormatException e) {
                weight = 0.0;
            }

            calculate();
        }

        @Override
        public void afterTextChanged(Editable s) { }

        @Override
        public void beforeTextChanged(
                CharSequence s, int start, int count, int after) { }
    };

    private final TextWatcher ageEditTextWatcher = new TextWatcher() {
        @Override
        public void onTextChanged(CharSequence s, int start,
                                  int before, int count) {
            try {
                age = Integer.parseInt(s.toString());
            }
            catch (NumberFormatException e) {
                age = 0;
            }

            calculate();
        }

        @Override
        public void afterTextChanged(Editable s) { }

        @Override
        public void beforeTextChanged(
                CharSequence s, int start, int count, int after) { }
    };


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/zdrowie/java/ui/kcal/KcalFragment.kt
package com.example.zdrowie.java.ui.kcal

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.widget.RadioGroup
import android.widget.RadioButton
import android.text.TextWatcher
import android.text.Editable
import android.view.View
import androidx.fragment.app.Fragment
import com.example.zdrowie.databinding.FragmentKcalBinding
import java.lang.NumberFormatException
import java.text.DecimalFormat

class KcalFragment : Fragment() {
    private var binding: FragmentKcalBinding? = null
    private var height = 0.0
    private var weight = 0.0
    private var age = 0
    private var sex = ""
    private var kcal = 0.0
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = FragmentKcalBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        binding!!.heightEditNumber.addTextChangedListener(heightEditTextWatcher)
        binding!!.weightEditNumber.addTextChangedListener(weightEditTextWatcher)
        binding!!.ageEditNumber.addTextChangedListener(ageEditTextWatcher)
        val sexRadioGroup = binding!!.sexRadioGroup
        sexRadioGroup.setOnCheckedChangeListener { _: RadioGroup?, _: Int ->
            val checkedButton =
                requireView().findViewById<View>(sexRadioGroup.checkedRadioButtonId) as RadioButton
            when (checkedButton.text.toString()) {
                "Female" -> sex = "female"
                "Male" -> sex = "male"
            }
            calculate()
        }
        return root
    }

    private fun calculate() {
        if (weight > 0 && height > 0 && age > 0 && (sex == "female" || sex == "male")) {
            kcal = if (sex == "male") {
                66.47 + 13.7 * weight + 5.0 * height - 6.76 * age
            } else {
                655.1 + 9.567 * weight + 1.85 * height - 4.68 * age
            }
        }
        setText(decimalFormat.format(kcal))
    }

    private fun setText(text: String?) {
        binding!!.bmrValueTextView.text = text
    }

    private val heightEditTextWatcher: TextWatcher = object : TextWatcher {
        override fun onTextChanged(
            s: CharSequence, start: Int,
            before: Int, count: Int
        ) {
            height = try {
                s.toString().toDouble()
            } catch (e: NumberFormatException) {
                0.0
            }
            calculate()
        }

        override fun afterTextChanged(s: Editable) {}
        override fun beforeTextChanged(
            s: CharSequence, start: Int, count: Int, after: Int
        ) {
        }
    }
    private val weightEditTextWatcher: TextWatcher = object : TextWatcher {
        override fun onTextChanged(
            s: CharSequence, start: Int,
            before: Int, count: Int
        ) {
            weight = try {
                s.toString().toDouble()
            } catch (e: NumberFormatException) {
                0.0
            }
            calculate()
        }

        override fun afterTextChanged(s: Editable) {}
        override fun beforeTextChanged(
            s: CharSequence, start: Int, count: Int, after: Int
        ) {
        }
    }
    private val ageEditTextWatcher: TextWatcher = object : TextWatcher {
        override fun onTextChanged(
            s: CharSequence, start: Int,
            before: Int, count: Int
        ) {
            age = try {
                s.toString().toInt()
            } catch (e: NumberFormatException) {
                0
            }
            calculate()
        }

        override fun afterTextChanged(s: Editable) {}
        override fun beforeTextChanged(
            s: CharSequence, start: Int, count: Int, after: Int
        ) {
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    companion object {
        private val decimalFormat = DecimalFormat("0.0")
    }
}